hack/check/misspell.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
+
+# Copyright (c) 2021 Tigera, Inc. All rights reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+
+CALICO_REPO_PATH="$(git rev-parse --show-toplevel)"
+
+MISSPELL_LOC="${CALICO_REPO_PATH}/hack/check/tools/bin"
+
+# Install tools we need if it is not present
+if [[ ! -f "${MISSPELL_LOC}/misspell" ]]; then
+  curl -L https://git.io/misspell | bash
```

Comment:
I don't think we should be piping shell scripts to bash; we have no idea who owns that URL redirect.  They could swap it to a bitcoin minor or worse after this PR goes in.

Our builds are containerised, if you want a new tool it should be added to the go-build container and run through that.